### PR TITLE
OpenBSD: add sendmmsg and recvmmsg support.

### LINKED ISSUE #3696: Add `recvmmsg` and `sendmmsg` on OpenBSD (*-unknown-openbsd)
`recvmmsg` is supported on OpenBSD since OpenBSD 7.2: https://www.openbsd.org/72.html
Same for `sendmmsg`.

Documentation is here:
- https://man.openbsd.org/OpenBSD-7.2/recvmmsg
- https://man.openbsd.org/OpenBSD-7.2/sendmmsg

Related PR made at the time when OpenBSD did not support these functions yet: https://github.com/rust-lang/libc/pull/1231

### PR DESCRIPTION
**Pull request by devnexen**
close #3696

### CODE REVIEW

**Review by semarie — CHANGES_REQUESTED** [review]
it doesn't pass the test suite under OpenBSD.

```
  cargo:warning=/data/semarie/repos/rust/build-libc/build/x86_64-unknown-openbsd/debug/build/libc-test-29cac031f36c8215/out/main.c:15544:51: error: must use 'struct' tag to refer to type 'mmsghdr'
  cargo:warning=             int (*__test_fn_sendmmsg(void))(int, mmsghdr*, unsigned int, int) {
  cargo:warning=                                                  ^
  cargo:warning=                                                  struct
  cargo:warning=/data/semarie/repos/rust/build-libc/build/x86_64-unknown-openbsd/debug/build/libc-test-29cac031f36c8215/out/main.c:15549:51: error: must use 'struct' tag to refer to type 'mmsghdr'
  cargo:warning=             int (*__test_fn_recvmmsg(void))(int, mmsghdr*, unsigned int, int, struct timespec*) {
  cargo:warning=                                                  ^
  cargo:warning=                                                  struct
  cargo:warning=2 errors generated.

  --- stderr
  rust version: 1.80.0


  error occurred: Command "/usr/local/bin/sccache" "cc" "-O0" "-ffunction-sections" "-fdata-sections" "-fPIC" "-gdwarf-2" "-fno-omit-frame-pointer" "-m64" "--target=x86_64-unknown-openbsd" "-Wall" "-Wextra" "-Wall" "-Wextra" "-Werror" "-Wno-unused-parameter" "-Wno-type-limits" "-Wno-address-of-packed-member" "-Wno-unknown-warning-option" "-Wno-deprecated-declarations" "-Wno-deprecated-declarations" "-o" "/data/semarie/repos/rust/build-libc/build/x86_64-unknown-openbsd/debug/build/libc-test-29cac031f36c8215/out/08e68c28a9818392-main.o" "-c" "/data/semarie/repos/rust/build-libc/build/x86_64-unknown-openbsd/debug/build/libc-test-29cac031f36c8215/out/main.c" with args cc did not execute successfully (status code exit status: 1).
```

**Review by semarie — APPROVED** [review]
LGTM now (but getnameinfo break the build)

**Comment by tgross35** [comment]
This is follow up of https://github.com/rust-lang/libc/pull/3703

File: libc-test/semver/openbsd.txt [hunk 1]
```diff
@@ -1210,6 +1210,7 @@ readlinkat
 reallocarray
 reboot
 recvmsg
+recvmmsg
```
Comment (tgross35) [inline libc-test/semver/openbsd.txt:1213]: Could you add `sendmmsg` and `mmsghdr`?

**Comment by tgross35** [comment]
OpenBSD and NetBSD actually seem to have a slightly different signature for `sendmmsg`: on OpenBSD the `mmsg` pointer is `const`, but not for NetBSD. I assume NetBSD could also be `const` even though this isn't in the API.

This doesn't seem worth special casing on, leaving it `mut` does work for both. So LGTM, thanks!

- https://man.openbsd.org/sendmsg.2
- https://man.netbsd.org/NetBSD-7.0/sendmmsg.2.

**Pull request merged**